### ISSUE TITLE
[Feature Request] Backlog management

### ISSUE THREAD

**Issue by vsheg** [open-spaced-repetition/fsrs4anki-helper#71]
Hi, it would be really useful if FSRS could manage backlogs, this feature is completely missing in Anki. 

## Backlog

Even if one has a straight review history, after a few missing days, the review count leads to frustration and could even push someone to quit Anki.

For a deck with a compete course ("Physics", "Chemistry", "Statistics", "Linear Algebra", "Calculus", etc.) the order of the cards in the deck is **very important**, so after a major break one has to review all the due cards in the **order they were created**.

> I have experienced this several times, and now I want to retell many cards (7000+) in "archive" decks that I studied years ago. I don't want to spend more that 30% of my usual review time dealing with the backlog, also I want to add new cards to this overdue decks and study it in parallel.

## Known strategies

1. **Forget** all the overdue cards. This won't work well, because in a normal case one still remembers 50-70% of this cards. If one uses "Forget"-option, one will completely lose the progress.
    > I have already measured my true retention on a small random subset of these cards, and it is about 60%, definitely not 0.
2. Create a bunch of **filtered decks** and review cards in the order they were added. Filtered decks come in handy when the backlog and the collection are not huge, they are useful for cram and ad hoc ordering, but they require manual setting and I have seen bug reports associated with their scheduling.

## Hypothetical solution

It would be good if Anki's regular decks scheduler allowed one to review cards in the order they were added. I don't know why filtered decks can do that and regular decks can't. But this is an issue for the Anki developers. 

## Worked before solution

Before, Anki had a "Reschedule" (I guess it was called that) option, it allowed one to select any (e.g. all overdue) cards sorted in any order (e.g. order added) and reschedule them for the next few months.

One can estimate the realistic number of overdue cards that can be reviewed each day (e.g. 30) and even continue the progress by adding and introducing completely new cards in parallel. Backlog and new stuff won't interfere even if they are in the same deck. No filtered deck, no manual setup, just an ordinary Anki routine. As a result, a few months later everything is reviewed and properly scheduled.

I used this before the V3 scheduler was introduced, now "Reschedule" option is missing.

Now Anki has

- "Set Due Date", which can set due date to a specific date for all cards at once, or to mix up all the reviews and completely miss the order (I use this for vocabulary decks, but it is useless for well structured decks);
- "Reposition", which only works with new cards.

## Feature request

It would be really useful if FSRS helper could return "Reschedule" option back to Anki, I guess it's manageable on a custom scheduling side and doesn't require changing existing Anki scheduler. If you could add an option that allows to reschedule a bunch of sorted (in browser) cards for a next N days evenly or considering a limit X cards/day it would be really beneficial for many users.

---

By the way, thanks for your work!

**Comment by L-M-Sherlock** [comment]
FSRS helper has the feature `postpone`. Did you use it?

**Comment by vsheg** [comment]
@L-M-Sherlock yes, I did. It doesn't preserve the original creation order, does it?

**Comment by L-M-Sherlock** [comment]
Does original creation order matter? I don't think so. The retention is most important in my review.

**Comment by L-M-Sherlock** [comment]
So the interval is not important in this case. FSRS just rescheduled these cards by their creation order. I think this feature is interesting, and the FSRS memory model adeptly manages reviews that deviate from the normal schedule.

**Comment by L-M-Sherlock** [comment]
So, what is the input of the feature? What context should this feature place?

A review limit per day is required. And a set of cards waiting for rescheduling.

By the way, I need to give this feature a new name. "Reschedule" has been used.

**Comment by L-M-Sherlock** [comment]
> The input is a set of cards (selected in the browser, maybe the whole deck). It might be located in the browser menu bar "Cards > Backlog Management", then users provide amount of cards/day they are able to review from the ground up.

I haven't implemented any features like this. I'm grateful if you could provide some examples implemented in other add-ones.

**Comment by L-M-Sherlock** [comment]
inactive
